pkg/controllers/dnshealthcheckprobe/dnshealthcheckprobe_controller.go

```diff
@@ -77,6 +78,21 @@ func (r *DNSHealthCheckProbeReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	probeId := probeId(previous)
 
+	additionalHeaders := v1alpha1.AdditionalHeaders{}
+
+	if previous.Spec.AdditionalHeaders != nil {
+		secretKey := client.ObjectKey{Name: previous.Spec.AdditionalHeaders.Name, Namespace: previous.Spec.AdditionalHeaders.Namespace}
```

Comment:
a namespaced resource shouldn't reference secrets in a different namespace, maybe we should just drop the `namespace` field and use the same namespace where the dnshealthcheckprobe lives?